### PR TITLE
Hashcons

### PR DESCRIPTION
**Pull request by AltGr**
Small fixes + add some trivial hash-consing for strings

### CODE REVIEW

File: src/opamLexer.mll [hunk 1]
```diff
@@ -111,6 +111,7 @@ rule token = parse
 | "("    { LPAR }
 | ")"    { RPAR }
 | '\"'   { STRING (buffer_rule string lexbuf) }
+| "\"\"\"\n" { STRING (buffer_rule string_triple lexbuf) }
```
Comment (dra27) [inline src/opamLexer.mll]: Is this correct? I thought that the point with #18 was that we should neither print nor ignore a newline at the start?

Comment (rjbou) [reply src/opamLexer.mll]: ping @AltGr

Comment (AltGr) [reply src/opamLexer.mll]: Hm, maybe a stale fix that I had laying around concurrently to #18 ? I don't remember :thinking:

File: src/opamLexer.mll [hunk 2]
```diff
@@ -81,13 +81,19 @@ let char_for_hexadecimal_code lexbuf i =
              else d2 - 48 in
   Char.chr (val1 * 16 + val2)
 
+(* Some hash-consing for strings *)
+module HS =
+  Weak.Make(struct include String let hash = Hashtbl.hash let equal = (=) end)
+let hm = HS.create 317
```
Comment (redianthus) [inline src/opamLexer.mll:87]: IIRC, nowadays it's better to use a power of 2.

**Review by rjbou — APPROVED** [review]

**Pull request merged**